TheMET/TheMET/Network/NetworkManager.swift

```diff
@@ -9,7 +9,20 @@ import Foundation
 
 class NetworkManager {
     
+    private var lastRequestTime: Date = .now
+    
     func get(urlString: String, parameters: [String : String], completion: @escaping ([String: Any]?) -> Void)  {
+        if self.isTimeIntervalLongEnough(lastRequest: lastRequestTime) {
+            self.executeNetworkGet(urlString: urlString, parameters: parameters, completion: completion)
+        } else {
+            DispatchQueue.global().asyncAfter(deadline: DispatchTime.now() + .milliseconds(Int(12.5)), qos: .background) {
+                self.executeNetworkGet(urlString: urlString, parameters: parameters, completion: completion)
+            }
+        }
+        lastRequestTime = .now
```

Comment:
The `lastRequestTime` should be updated in `executeNetworkGet` method when an actual request is sent. Otherwise you would just shift time with no good reason.